packages/react-ui/src/hooks/useImperativeIntl.js

```diff
@@ -9,21 +9,36 @@ import {
 
 const cache = createIntlCache();
 
-export default function useImperativeIntl(intlConfig) {
-  return useMemo(() => {
+let lastIntlConfig;
+let cachedC4rIntl;
```

Comment:
Optional — For me this would be a little more readable if we omit the "C4r" abbreviation and align the two names. Perhaps:

```suggestion
let cachedIntlConfig;
let cachedIntl;
```